### PR TITLE
APR-1444 add question data API endpoint for criminal compliance pages

### PR DESCRIPTION
**Pull request by David-B-Read**

### CODE REVIEW

File: src/SFA.DAS.ApplyService.InternalApi/Controllers/RoatpGatewayCriminalComplianceChecksController.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                            if (furtherInformationAnswer != null)
+                            {
+                                criminalComplianceCheckDetails.FurtherQuestionId = furtherInformationAnswer.QuestionId;
+                                criminalComplianceCheckDetails.FurtherAnswer = furtherInformationAnswer.Value;
+                                break;
+                            }
+                        }
+                    }
+                }
+            }
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.ApplyService.InternalApi/Controllers/RoatpGatewayCriminalComplianceChecksController.cs]: I think this block would benefit from a little bit of refactoring, maybe just extract a few methods to make it obvious at a glance what is going on?

File: src/SFA.DAS.ApplyService.InternalApi/Services/CriminalComplianceChecksQuestionLookupService.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+using SFA.DAS.ApplyService.InternalApi.Models.Roatp;
+using System;
+
+namespace SFA.DAS.ApplyService.InternalApi.Services
+{
+    public class CriminalComplianceChecksQuestionLookupService : ICriminalComplianceChecksQuestionLookupService
+    {
+        public QnaQuestionDetails GetQuestionDetailsForGatewayPageId(string gatewayPageId)
+        {
+            switch (gatewayPageId)
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.ApplyService.InternalApi/Services/CriminalComplianceChecksQuestionLookupService.cs]: Whenever I see a switch I get nervous about how the code is going to end up.  As there is only 1 path I'd say this doesn't need a switch statement for now so we should remove it and look at alternative patterns as and when it grows.

Comment (David-B-Read) [reply src/SFA.DAS.ApplyService.InternalApi/Services/CriminalComplianceChecksQuestionLookupService.cs]: Currently only one, yes. The intent is for the same view/view model to be re-used for all the criminal compliance questions as they all have the same format. This is a lookup from gateway id to the QnA data it needs to lookup to populate the page

File: src/SFA.DAS.ApplyService.InternalApi/Controllers/RoatpGatewayCriminalComplianceChecksController.cs (hunk 1 above)
Comment (David-B-Read) [reply src/SFA.DAS.ApplyService.InternalApi/Controllers/RoatpGatewayCriminalComplianceChecksController.cs]: Updated so it's clearer what the inner loop is for

**Pull request closed without merging**